src/inspect.rs

```diff
@@ -13,6 +13,44 @@ use crate::cli::*;
 use crate::errors::CliError::*;
 use crate::input::*;
 
+fn handle_query(
+    query: &Rule,
+    query_all: bool,
+    all_params: &[Param],
+    authorizer: &mut Authorizer,
+) -> Result<()> {
+    let mut rule = query.clone();
+
+    for p in all_params {
+        match p {
+            Param::Term(name, t) => {
+                rule.set_lenient(name, t)?;
+            }
+            Param::PublicKey(name, pk) => {
+                rule.set_scope_lenient(name, *pk)?;
+            }
+        }
+    }
+
+    let facts: Vec<Fact> = if query_all {
+        authorizer.query_all(rule.clone())?
+    } else {
+        authorizer.query(rule.clone())?
+    };
+
+    println!();
+    println!("🔎 Running query: {}", &rule);
```

Comment:
Why using println directly (not only in this case) and not some logger?

Reply:
this is a CLI tools, whose purpose is to output human-readable text on stdout, so it's one of the rare cases where using println directly is ok.

There is an open issue for JSON support #10, which will force us to remove all the printlns littered in the code and move all output generation to a single place.